packages/core/src/motify.tsx

```diff
@@ -1,32 +1,31 @@
-import React, { ComponentType, forwardRef } from 'react'
-import type { ImageStyle, TextStyle, ViewStyle } from 'react-native'
+import React, {
+  Component,
+  ComponentType,
+  forwardRef,
+  PropsWithChildren,
+} from 'react'
+import type { StyleProp } from 'react-native'
 import type { MotiProps } from './types'
 import useMapAnimateToStyle from './use-map-animate-to-style'
 import Animated from 'react-native-reanimated'
 
 // https://www.framer.com/blog/posts/magic-motion/
 export default function motify<
-  Style,
-  Props extends { style?: Style },
-  Ref,
-  ExtraProps,
-  // Variants,
-  // Animate = ViewStyle & TextStyle
-  Animate = ViewStyle | ImageStyle | TextStyle
+  Props extends { style?: object | StyleProp<any> },
+  Ref = Component<Props>,
+  Style = Props['style'] extends StyleProp<infer S> ? S : Record<string, any>
```

Comment:
Question, should this be something like:

```ts
Style = Props['style'] extends StyleProp<infer S> ? S : Props['style'] extends Record<string, any> ? Props['style'] : Record<string, any>
```

I'm not sure if that exact code works, but the idea is: first check if it's a style prop, otherwise check if it's a custom `style` type that matches an object, and otherwise, make it a plain object.

It's possible a component will have a custom `style` prop that doesn't match `StyleProp`, so I want to make sure those types aren't lost necessarily.